net/tshttpproxy/tshttpproxy_synology.go

```diff
@@ -0,0 +1,137 @@
+// Copyright (c) 2022 Tailscale Inc & AUTHORS All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+//go:build linux
+// +build linux
+
+package tshttpproxy
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"net"
+	"net/http"
+	"net/url"
+	"os"
+	"strings"
+	"sync"
+	"time"
+
+	"tailscale.com/util/lineread"
+	"tailscale.com/version/distro"
+)
+
+func init() {
+	sysProxyFromEnv = linuxSysProxyFromEnv
+}
+
+var (
+	distroOnce         sync.Once
+	distroProxyFromEnv func(*http.Request) (*url.URL, error)
+)
+
+func linuxSysProxyFromEnv(req *http.Request) (*url.URL, error) {
+	distroOnce.Do(func() {
+		if distro.Get() == distro.Synology {
+			distroProxyFromEnv = synologyProxyFromConfigCached
+			return
+		}
+	})
+	if distroProxyFromEnv != nil {
+		return distroProxyFromEnv(req)
+	}
+	return nil, nil
+}
+
+var cache struct {
+	sync.Mutex
+	proxy   *url.URL
+	updated time.Time
+}
+
+func synologyProxyFromConfigCached(req *http.Request) (*url.URL, error) {
+	cache.Lock()
+	defer cache.Unlock()
+
+	if !cache.updated.IsZero() && time.Since(cache.updated) < time.Minute {
```

Comment:
I can see users getting confused that they changed their synology settings and then it doesn't work for a minute.

Isn't it cheap enough to do a Lstat of the file per call? We're about to make a network request, after all. A local lstat to the dentry cache should be lost in the noise.

Reply:
I'd need to do a full stat, and now done.. I tend to be untrusting of vendors that they've applied appropriate flags to avoid flash wear - just paranoia not wanting to wear peoples devices out prematurely.